google/downloader.sh

```diff
@@ -6,58 +6,100 @@
 # From: https://gist.github.com/jeffmccune/e7d635116f25bc7e12b2a19efbafcdf8
 # From: https://gist.github.com/n0531m/f3714f6ad6ef738a3b0a
 
+# Script to retrieve and organize Google and Google Cloud IP ranges.
+
 set -euo pipefail
 set -x
 
+# Check for required dependencies
+for cmd in curl dig jq mktemp; do
+  if ! command -v "$cmd" &> /dev/null; then
+    echo "Error: $cmd is not installed or not in PATH" >&2
+    exit 1
+  fi
+done
+
+# Create a temporary directory and ensure cleanup on exit
+temp_dir=$(mktemp -d)
+trap 'rm -rf -- "$temp_dir"' EXIT
 
+# Function to download files with retries and error handling
+download_file() {
+  local url=$1
+  local output_file=$2
+  local retries=3
+  local count=0
+  until curl -s "$url" -o "$output_file"; do
+    count=$((count + 1))
+    if [[ $count -ge $retries ]]; then
+      echo "Error: Failed to download $url after $retries attempts"
+      exit 1
+    fi
+    sleep 2  # wait before retrying
+  done
+}
 
-# get from public ranges
-curl -s https://www.gstatic.com/ipranges/goog.txt > /tmp/goog.txt
-curl -s https://www.gstatic.com/ipranges/cloud.json > /tmp/cloud.json
+# Parallel downloads with retries
+download_file "https://www.gstatic.com/ipranges/goog.txt" "$temp_dir/goog.txt" &
+download_file "https://www.gstatic.com/ipranges/cloud.json" "$temp_dir/cloud.json" &
+download_file "https://developers.google.com/search/apis/ipranges/googlebot.json" "$temp_dir/googlebot.json" &
+wait  # Ensure all downloads finish
 
-# Public GoogleBot IP ranges
-# From: https://developers.google.com/search/docs/advanced/crawling/verifying-googlebot
-curl -s https://developers.google.com/search/apis/ipranges/googlebot.json > /tmp/googlebot.json
+# Fetch Google netblocks using dig command
+fetch_netblocks() {
+  local idx=2
+  local txt
+  txt="$(dig TXT _netblocks.google.com +short @8.8.8.8 || true)"
```

Comment:
`|| true` will hide network problems from us that we might have noticed, in which case we just won't know anything?

Reply:
Uhm.. maybe the best can be to collect errors as artifacts for further investigation.. but as you already said.. it works and I cimpletely agree with you, keep this PR as my thanks message for your immensely useful repo. I use it every day since years :)